tests/integration/pkg/helpers/api_rule.go

```diff
@@ -123,13 +124,13 @@ func ApplyApiRuleV2Alpha1ExpectError(toExecute RetryableApiRule, onRetry Retryab
 			if err != nil {
 				return err
 			}
-			if apiStatus.Status.State != "Error" {
-				log.Println("APIRule status not Error: " + apiStatus.Status.Description)
-				return errors.New("APIRule status not Error: " + apiStatus.Status.Description)
+			if apiStatus.Status.State != status {
+				log.Printf("APIRule status not %s: %s", status, apiStatus.Status.Description)
+				return fmt.Errorf("APIRule status not %s: %s", status, apiStatus.Status.Description)
 			}
-			if !strings.Contains(apiStatus.Status.Description, errorMessage) {
-				log.Println("APIRule Error status description does not contain expected string: " + apiStatus.Status.Description)
-				return errors.New("APIRule Error status description does not contain expected string: " + apiStatus.Status.Description)
+			if descMessage != "" && !strings.Contains(apiStatus.Status.Description, descMessage) {
+				log.Printf("APIRule Status description does not contain expected string: %s", apiStatus.Status.Description)
+				return fmt.Errorf("APIRule Status description does not contain expected string: %s", apiStatus.Status.Description)
```

Comment:
Error should not be capitalized https://go.dev/wiki/CodeReviewComments#error-strings